libsel4/include/sel4/bootinfo_types.h

```diff
@@ -75,11 +76,23 @@ typedef struct seL4_BootInfo {
      * to make this struct easier to represent in other languages */
 } seL4_BootInfo;
 
-/* If extraLen > 0, then 4K after the start of bootinfo there is a region of the
- * size extraLen that contains additional boot info data chunks. They are
- * arch/platform specific and may or may not exist in any given execution. Each
- * chunk has a header that contains an ID to describe the chunk. All IDs share a
- * global namespace to ensure uniqueness.
+/* The fixed boot info frame size is one page. As the page size is 4 KiByte on
+ * x86, ARM and RISC-V, lots of userland code has a hard coded assumption that
+ * the boot info frame size is 4 KiByte, but it should use SEL4_BI_FRAME_SIZE
+ * actually.
```

Comment:
```suggestion
/* The fixed boot info frame size, per default 4 KiByte on x86, Arm and RISC-V.
 * Instead of assuming the default, userland code should use the SEL4_BI_FRAME_SIZE constant.
```